### ISSUE TITLE
[BUG] Logo paths are incorrect when serving under a subfolder

### ISSUE THREAD

**Issue by doganulus** [gethinode/hinode#727]
**Describe the bug**
When serving a Hinode site under a subfolder such as `organization.github.io/repo_name`, navbar logos do not appear correctly because image paths are incorrect.

**To reproduce**
1. Serve the site under a subfolder
2. Configure logo for light and dark modes

**Expected behavior**
- Logos appear correctly when serving under a subfolder
- Subfolder base URLs are fully supported in layout templates

**Environment (please complete the following information):**
 - Hinode version: v0.22.3

**Additional context**
I needed to override `navbar` partial to prepend `site.Home.RelPermalink` for `logoLight` and `LogoDark` paths for a hacky solution as [here](https://github.com/bouncmpe/rv24/blob/dc2b47f612545e6b8dddde03052ef685473949f8/layouts/partials/assets/navbar.html#L193).

**Comment by doganulus** [comment]
My repo is here: https://github.com/bouncmpe/rv24

The error occurred locally and on the server before I hacked the `navbar` partial layout to handle logo URLs correctly under a subfolder. 

This problem occurs because those paths start from `/` rather than `/subfolder/`.

**Comment by markdumay** [comment]
I was able to reproduce your error. I have no experience with deploying to a subfolder, but I found this [thread](https://discourse.gohugo.io/t/make-home-to-be-subdirectory/4345/6) in the Hugo discussion forum that addresses a similar issue.

Following the recommendations in the discussion thread, I disabled the custom `navbar.html` partial and added the following settings to `hugo.toml`:

```toml
baseURL = "//example.com/rv24/"
canonifyURLs = true
```

The logo renders correctly on my local machine now. However, I haven't tested this in production yet. The `example.com` is a dummy value, but seems required in order for Hugo to pick up the subfolder as expected. It's worth giving a try?

If it works, you could replace the baseURL with your actual deployment url. Possibly you would need to differentiate for local development and production deployment. You could override the `baseURL` in `config/production/hugo.toml` to do so.

**Comment by markdumay** [comment]
Glad to hear it works! I agree it's not an ideal solution. I remember I ran into a similar issue when adding support for [versioning](https://gethinode.com/guides/versioning/#configuring-the-multi-branch-deployment). I added a custom setting specifically for the retrieval of assets:

```toml
[main]
    canonifyAssetsURLs = true
```

I haven't tested if this works for the navbar icon assets too, but it worth trying?

**Comment by oupala** [comment]
I encountered the same issue as described here. And it took me **weeks** to find out a workaround.

And the workaround is not a solution but a real workaroud, that uses a deprecated function that is likely to be removed in a future version.

I used many other static generators and it it the first time that I get blocked by a such simple issue.

I think this deserves a real solution, not a weak workaround.
